### PR TITLE
[Feature] Adding interactive tables

### PR DESCRIPTION
**Pull request by hjoaquim**
# Pull Request Template for OpenBB Developers

1. **Why**? (1-3 sentences or a bullet point list):

    - Allowing the user to have interactive tables.

2. **What**? (1-3 sentences or a bullet point list):

    - Adds a method to the charting extension that will display the interactive table (depending on `backend.isatty`)
    - Fixes the table's html.

3. **Impact** (1-2 sentences or a bullet point list):

    - Doesn't affect the current functionality of the charting extension.

4. **Testing Done**:

    - Display tables
       - `isatty=True`
image

       -  `isatty=False`
image

### CODE REVIEW

**Review by IgorWounds — APPROVED** [review]

**Review by deeleeramone — CHANGES_REQUESTED** [review]
This will be great! There's a few items I've run into, a couple I'll outline here because it will blow up GitHub by commenting in table.html.

- When dates get converted to JSON, we end up with all this extra noise that makes the table much harder to read, and columns way bigger than they should be. Dates should remain dates and not get encoded with time and TZ information. Converting the column to a string instead of using the `iso` date output of `to_json` will help.

Screenshot 2024-04-05 at 1 21 33 PM

- Somewhere in there is a process which automatically format columns with "date-like" names, but this produces undesirable results. Perhaps kill that entirely?

Screenshot 2024-04-05 at 1 13 46 PM

In the V3 implementation there is a parameter for a list of columns to preserve types so that it doesn't convert it. That was a very useful fallback for when numbers 

Do you know if there's a way to fit contents based on cell (instead of header) on creation? Otherwise we need to manually move each column around to get something you can read.  


Screenshot 2024-04-05 at 1 34 18 PM

Screenshot 2024-04-05 at 1 09 02 PM

**Comment by deeleeramone** [comment]
I get an empty window when I use this function:

```
obb.equity.fundamental.ratios("AAPL")
```

The devtools shows: Invalid character '\u20ac'

File: openbb_platform/obbject_extensions/charting/openbb_charting/__init__.py [hunk 1]
```diff
@@ -51,7 +54,7 @@ def __init__(self, obbject):
             user_settings=self._obbject._user_settings,  # type: ignore
             system_settings=self._obbject._system_settings,  # type: ignore
         )
-        self._handle_backend()
+        self._backend: Backend = self._handle_backend()
```
[the remark this answers is not in the document]
Comment (hjoaquim) [reply openbb_platform/obbject_extensions/charting/openbb_charting/__init__.py:58]: I don't agree - backend is treated as a global variable and will only get instantiated once. Also, I think it's a good idea to keep its initialization as centralized as possible.

File: openbb_platform/obbject_extensions/charting/openbb_charting/__init__.py [hunk 2]
```diff
@@ -212,3 +221,39 @@ def to_chart(
...
+        ----------
+        data : Optional[Union[pd.DataFrame, pd.Series]], optional
+            Data to be plotted, by default None.
+            If no data is provided the OBBject results will be used.
+        """
+        data_as_df, _ = self._prepare_data_as_df(data)
+        if self._backend.isatty:
+            try:
+                self._backend.send_table(
+                    df_table=data_as_df,
```
[the remark this answers is not in the document]
Comment (hjoaquim) [reply openbb_platform/obbject_extensions/charting/openbb_charting/__init__.py]: Nice catch! Done here https://github.com/OpenBB-finance/OpenBBTerminal/pull/6290/commits/64cb679da13a7f567ba97828ad08da4bef62aab7

[the remark this answers is not in the document]
Comment (hjoaquim) [reply openbb_platform/obbject_extensions/charting/openbb_charting/__init__.py]: title defaults to the route - but can be changed by using the `title` argument

**Comment by hjoaquim** [comment]
> I get an empty window when I use this function:
> 
> ```
> obb.equity.fundamental.ratios("AAPL")
> ```
> 
> The devtools shows: Invalid character '\u20ac'

looks like that's fixed now!

image

**Comment by hjoaquim** [comment]
@deeleeramone if you agree, I'll ask for a stamp here and will add the other comments to our backlog as this is needed for the CLI, and we can address those bugs later without prejudice (imo)

**Comment by deeleeramone** [comment]
> if you agree, I'll ask for a stamp here and will add the other comments to our backlog as this is needed for the CLI, and we can address those bugs later without prejudice (imo)

We can definitely iterate on this going forward, but I let's nip this one in the butt first because it's such a glaring issue:
Screenshot 2024-04-08 at 9 35 06 AM

I might have an idea to patch that quickly.

**Comment by deeleeramone** [comment]
@hjoaquim, could of cheap and cheerful, temporary, solutions here. This React/Tailwinds stuff is just awful to work with.

Screenshot 2024-04-08 at 2 41 37 PM

**Review by deeleeramone — APPROVED** [review]

**Comment by hjoaquim** [comment]
Thanks a lot for shipping in @deeleeramone :butterfly:

**Pull request merged**